src/main/java/md/miller1995/Dealership/controllers/CarController.java

```diff
@@ -22,11 +21,19 @@ public CarController(CarService carService){
         this.carService = carService;
     }
 
-    @GetMapping("/{id}")
+    @GetMapping("/search/by/{id}")
     public ResponseEntity<CarDTO> getCarById(@PathVariable(name = "id") long id){
         CarEntity carEntity = carService.findById(id);
         CarDTO carDTO = carService.carEntityToCarDTO(carEntity);
 
         return new ResponseEntity<CarDTO>(carDTO, HttpStatus.OK);
     }
+
+    @GetMapping("/search/by")
+    public ResponseEntity<List<CarDTO>> findCarByModel(@RequestParam(value = "model") String model){
+        List<CarEntity> carEntityList = carService.findByModel(model);
+        List<CarDTO> carDTOList = carService.carEntityListToCarDTOList(carEntityList);
+
+        return new ResponseEntity<List<CarDTO>>(carDTOList, HttpStatus.OK);
```

Comment:
- all the logic from lines 34 and 35 should be inside the carService
- service methods should be more descriptive to reflect intention; is not clear enough the fact that carService method findByModel returns a list of carEntities
- service methods should return DTOs, not entities

Reply:
- moved line 34 and 35 in service layer
- I renamed method in carService for to be more descriptive